idgenerator/aws/xray/aws_xray_idgenerator.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws // import "go.opentelemetry.io/contrib/idgenerator/aws/xray"
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"log"
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type spanID [8]byte
+type traceID [16]byte
+
+// IDGenerator is an interface for generating new TraceIDs and SpanIDs
+type IDGenerator interface {
+	NewTraceID() traceID
+	NewSpanID() spanID
+}
+
+type xRayIDGenerator struct {
+	sync.Mutex
+	randSource *rand.Rand
+}
+
+// XRayIDGenerator returns an idGenerator used for sending traces to AWS X-Ray
+func XRayIDGenerator() IDGenerator {
```

Comment:
Remove IDGenerator interface, rename xRayIDGenerator to IDGenerator, rename this to NewIDGenerator() *IDGenerator.